bftengine/src/bftengine/ReplicaImp.cpp

```diff
@@ -1897,9 +1897,15 @@ void ReplicaImp::onMessage<ReplicaStatusMsg>(ReplicaStatusMsg *msg) {
   const ReplicaId msgSenderId = msg->senderId();
   const SeqNum msgLastStable = msg->getLastStableSeqNum();
   const ViewNum msgViewNum = msg->getViewNumber();
-  ConcordAssertEQ(msgLastStable % checkpointWindowSize, 0);
```

Comment:
What's the justification behind this change? This seems like a pretty serious error to run into. The assert was added presumably because a violation of this invariant is a protocol bug.

The rationale for changes like this should be put in the commit message, as just listing the change is not sufficient for a reviewer.

Reply:
Hey Andrew, the idea is - this error happens in another replica (the sender). No need to crash the receiver because of this :)

Reply:
Yes, it is a serious problem, but it is most likely in the peer Replica. This leaves an open door for a Byzantine peer to bring all the Replicas down by reporting an inconsistent Stable Sequence ID. If we detect that a peer has probably gotten corrupted I think we must log an error and not shut ourself down?

Reply:
On a side note, this actually doesn't happen. We just noticed the vulnerability while reviewing the code in this area.

Reply:
Aha! This is an excellent point! Thank you for the clarification.

Reply:
Actually after researching deeper, I see that we have this check in the validate method of this message [ReplicaStatusMsg.cpp:103](https://github.com/vmware/concord-bft/blob/b81b12406a2f838cc8797bda7aa4413ccb7218e1/bftengine/src/bftengine/messages/ReplicaStatusMsg.cpp#L103). Basically this means that malformed StatusRequest messages will be discarded on an earlier stage -> [ReplicaBase.hpp:90](https://github.com/vmware/concord-bft/blob/b81b12406a2f838cc8797bda7aa4413ccb7218e1/bftengine/src/bftengine/ReplicaBase.hpp#L90).